action/mail.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * @license GNU General Public License, version 2
+ */
+
+
+if (!defined('DOKU_INC')) die();
+
+
+/**
+ * Class action_plugin_publish_mail
+ *
+ * @author Michael Große <grosse@cosmocode.de>
+ */
+class action_plugin_publish_mail extends DokuWiki_Action_Plugin {
+
+    /**
+     * @var helper_plugin_publish
+     */
+    private $hlp;
+
+    function __construct() {
+        $this->hlp = plugin_load('helper','publish');
+    }
+
+    public function register(Doku_Event_Handler $controller) {
+
+        $controller->register_hook('IO_WIKIPAGE_WRITE', 'AFTER', $this, 'send_change_mail', array());
+    }
+
+    /**
+     * send an email to inform about a changed page
+     *
+     * @param $event
+     * @param $param
+     * @return bool|mixed
+     */
+    function send_change_mail(&$event, $param) {
+        global $ID;
+        global $ACT;
+        global $INFO;
+        $data = pageinfo();
+
+        if ($ACT != 'save') {
+            return true;
+        }
+
+        // IO_WIKIPAGE_WRITE is always called twice when saving a page. This makes sure to only send the mail once.
+        if (!$event->data[3]) {
+            return true;
+        }
+
+        // Does the publish plugin apply to this page?
+        if (!$this->hlp->isActive($ID)) {
+            return true;
+        }
+
+        //are we supposed to send change-mails at all?
+        if (!$this->getConf('send_mail_on_change')) {
+            return true;
+        }
+
+        // get mail receiver
+        $receiver = $this->getConf('apr_mail_receiver');
+
+        // get mail sender
+        $sender = $data['userinfo']['mail'];
+
+        if ($sender == $receiver) {
+            dbglog('[publish plugin]: Mail not send. Sender and receiver are identical.');
+            return true;
+        }
+
+        if ($INFO['isadmin'] == '1') {
+            dbglog('[publish plugin]: Mail not send. Sender is admin.');
+            return true;
+        }
+
+        // get mail subject
+        $timestamp = $data['lastmod'];
+        $datum = dformat("d.m.Y",$timestamp);
+        $uhrzeit = dformat("H:i",$timestamp);
+        $subject = $this->getLang('apr_mail_subject') . ': ' . $ID . ' - ' . $datum . ' ' . $uhrzeit;
+        dbglog($subject);
```

Comment:
If you want to keep all the debug statements in, at least prefix them all with the same prefix ([publish plugin]).